### ISSUE TITLE
Errors using fences within tabs

### ISSUE THREAD

**Issue by fherreazcue** [carpentries/sandpaper#614]
When using a tab or group-tab (as implemented [here](https://github.com/carpentries/sandpaper/pull/571)), if a fence is used within, the content gets triplicated.
This is a minimal example to reproduce the error:
```
:::::::::::: tab

### tab 1

some text

### tab 2

some other text

:::::: spoiler

some text that should be hidden

::::::::::::::

::::::::::::::::::
```
And this is the output:

![image](https://github.com/user-attachments/assets/64bd5428-f3ea-4aa6-a7a0-927a68ce7631)

**Comment by astroDimitrios** [comment]
Hi both! @froggleston @fherreazcue This is a limitation of the lua, see the [caveats section on 571](https://github.com/carpentries/sandpaper/pull/571#issue-2122856009) in the first comment. I tried several ways to get fenced blocks to work inside the tabs but because of the way some of the admonitons are rendered as multiple inline elements there was no way to capture those properly inside a tab (although I'm still a Lua novice so maybe there is a better way).

Can the content you are writing be re-written to not include the hidden text? I can try and have another look at the Lua code but can't make any promises!

**Comment by matthewfeickert** [comment]
If it would be possible to reinvestigate if this is possible or not that would be useful as this is a limitation for writing multiplatform sections of lessons where there's `callout`s or `caution`s that only make sense in the context of a single platform.

**Comment by matthewfeickert** [comment]
Tab support is broken another way: You can only have one level of information in a tab structure &mdash; there's no way to have subheadings inside of a `tab` group.

Example:

````markdown
::: tab

### Unix machines (Linux and macOS)

```bash
curl -fsSL https://pixi.sh/install.sh | sh
```

or

```bash
wget -qO- https://pixi.sh/install.sh | sh
```

#### Autocompletions

##### Bash

Add the following to the end of `~/.bashrc`:

```bash
eval "$(pixi completion --shell bash)"
```

##### Zsh

Add the following to the end of `~/.zshrc`:

```zsh
autoload -Uz compinit && compinit  # redundant with Oh My Zsh
eval "$(pixi completion --shell zsh)"
```

##### Fish

Add the following to the end of `~/.config/fish/config.fish`:

```shell
pixi completion --shell fish | source
```

### Windows

Windows stuff

:::
````

produces 

![Image](https://github.com/user-attachments/assets/f4de7077-1069-4839-923d-dc0c9d4b2af2)

with all of the subsections on shell autocompletions not being rendered.

So as is, tabs need to be very simple and plain structures.

**Comment by froggleston** [comment]
Just checking in - @matthewfeickert @astroDimitrios is this still an issue with the latest changes to the group tabs headings in #709 ?

**Comment by matthewfeickert** [comment]
> is this still an issue with the latest changes to the group tabs headings in [#709](https://github.com/carpentries/sandpaper/pull/709) ?

@froggleston Yes. Testing with `sandpaper` `v0.20.0` my example from https://github.com/carpentries/sandpaper/issues/614#issuecomment-2973438899 renders out the `Autocompletions` subsections but not in a useful manner as it treats _everything_ in the `tab` at the same level.

<img width="1427" height="891" alt="Image" src="https://github.com/user-attachments/assets/4328d529-c22b-42ea-9358-535fd94f1ab2" />

**Comment by astroDimitrios** [comment]
Hi @froggleston, thanks for dealing with all the issues I seem to be creating with tabs! (At least they are being used!) I think there are now two issues here. The original issue where you can't use nested callouts in tabs (very difficult because of the way the lua filters work - still not been able to make it possible) and the issue from @matthewfeickert where you can't have subheadings in a tab.

Both of these raise questions about whether we want people to have the option of:

- nesting callouts in tabs (the original issue)
- using subheadings in tabs

Both increase the complexity of the content in the actual tab. Although having subheadings in tabs in theory is possible.
